src/PyNvCodec/src/PyNvEncoder.cpp

```diff
@@ -352,6 +374,66 @@ bool PyNvEncoder::EncodeSurface(shared_ptr<Surface> rawSurface,
   return EncodeSingleSurface(ctx);
 }
 
+bool PyNvEncoder::EncodeFromNVCVImage(py::object& nvcvImage,
+                                py::array_t<uint8_t>& packet)
+{
+  struct NVCVImageMapper {
+    int nWidth[2];
+    int nHeight[2];
+    int nStride[2];
+    CUdeviceptr ptrToData[2];
+  };
+
+  NVCVImageMapper nv12Mapper;
+
+  memset(&nv12Mapper, 0, sizeof(NVCVImageMapper));
+  py::list list = nvcvImage.attr("cuda")();
+
+  for (size_t i = 0; i < list.size(); i++) {
+    py::object nvcvImageType = list[i];
+
+    if (py::hasattr(nvcvImageType, "__cuda_array_interface__")) {
+      py::dict dict =
+          nvcvImageType.attr("__cuda_array_interface__").cast<py::dict>();
+      if (!dict.contains("shape") || !dict.contains("typestr") ||
+          !dict.contains("data") || !dict.contains("version")) {
+        return false;
+      }
+      int version = dict["version"].cast<int>();
+      if (version < 2) {
+        return false;
+      }
+
+      py::tuple dataPtr = dict["data"].cast<py::tuple>();
+      void* ptr = reinterpret_cast<void*>(dataPtr.cast<uint8_t>());
```

Comment:
Can you check whether 
```
              CUdeviceptr ptr;
              cuPointerGetAttribute(&ptr, CU_POINTER_ATTRIBUTE_DEVICE_POINTER, input_ptr);
```
is necessary? Numba documentation (https://numba.pydata.org/numba-doc/dev/cuda/cuda_array_interface.html, see documentation for `data`) recommends it for the case that pointer is coming from a different CUDA context.

We need to be in our CUDA context to perform most our operations. CVCUDA is always primary CUDA context?

Reply:
tracking this is new feature branch